### PR TITLE
Fix IndexSizeError by checking if audio is seekable, fix invalidHTMLMediaElement by first checking for source

### PR DESCRIPTION
**Pull request by jakemmarsh**
Add a check to first verify that the current audio is seekable before attempting to call `seekable.end(0)`. Fixes this error:

screen shot 2014-10-13 at 5 52 24 pm

Also adds a check to ensure that we only call `createMediaElementSource` if we don't already have a source, preventing the `invalid HTMLMediaElement` error.

### CODE REVIEW

File: dancer.js [hunk 1]
```diff
@@ -411,7 +411,9 @@
       }
 
       this.audio.addEventListener( 'progress', function ( e ) {
-        if ( e.currentTarget.duration ) {
+        var isSeekable = e.currentTarget.seekable && e.currentTarget.seekable.length > 0;
```
Comment (likethemammal) [inline dancer.js:414]: `e.currentTarget.seekable` could be a variable, since you're accessing it twice.

File: dancer.js [hunk 2]
```diff
@@ -524,7 +528,9 @@
       }, false);
 
       this.audio.addEventListener( 'progress', function ( e ) {
-        if ( e.currentTarget.duration ) {
+        var isSeekable = e.currentTarget.seekable && e.currentTarget.seekable.length > 0;
```
Comment (likethemammal) [inline dancer.js:531]: Same here with `e.currentTarget.seekable`

File: src/adapterMoz.js [hunk 3]
```diff
@@ -27,7 +27,9 @@
       }, false);
 
       this.audio.addEventListener( 'progress', function ( e ) {
-        if ( e.currentTarget.duration ) {
+        var isSeekable = e.currentTarget.seekable && e.currentTarget.seekable.length > 0;
```
Comment (likethemammal) [inline src/adapterMoz.js:30]: Same here

File: src/adapterWebAudio.js [hunk 4]
```diff
@@ -46,7 +46,9 @@
       }
 
       this.audio.addEventListener( 'progress', function ( e ) {
-        if ( e.currentTarget.duration ) {
+        var isSeekable = e.currentTarget.seekable && e.currentTarget.seekable.length > 0;
```
Comment (likethemammal) [inline src/adapterWebAudio.js:49]: Same here

**Comment by theunknownartisthour** [comment]
In this case it's best not to do look ahead (for / 0 type errors), just use fail-fast exception handling.
try{
_this.progress = e.currentTarget.seekable.end( 0 ) / e.currentTarget.duration;
} catch (e) {
_this.progress = 1;
}
This is a better implementation ;)

**Pull request closed without merging**